parsec/backend/memory/factory.py

```diff
@@ -59,10 +64,12 @@ async def _dispatch_event():
         "block": block,
         "blockstore": blockstore,
     }
-    for component in (organization, user, invite, message, realm, vlob, ping, block):
-        component.register_components(**components)
+    for component in components.values():
+        method = getattr(component, "register_components", None)
+        if method is not None:
+            method(**components)
 
-    async with trio.open_service_nursery() as nursery:
+    async with open_service_nursery() as nursery:
```

Comment:
why remove the `trio.` ? I'm not against it, but it seems to be the way we access `open_service_nursery` in the rest of the codebase

Reply:
Because it's added to the trio namespace with monkey patching, and trio doesn't like it.

> but it seems to be the way we access open_service_nursery in the rest of the codebase

Not in the `parsec.core.fs` module where mypy is enabled :)